packages/@sanity/desk-tool/src/panes/document/statusBar/sparkline/DocumentSparkline.tsx

```diff
@@ -1,74 +1,85 @@
-import {EditIcon} from '@sanity/icons'
 import {Box, Flex, useElementRect} from '@sanity/ui'
-import React, {useEffect, useMemo, useState, useRef, memo} from 'react'
-import {raf2} from '../../../../utils/raf'
+import React, {useEffect, useMemo, useState, memo, useLayoutEffect} from 'react'
+import {useSyncState} from '@sanity/react-hooks'
 import {useDocumentPane} from '../../useDocumentPane'
 import {DocumentBadges} from './DocumentBadges'
-import {ReviewChangesButton} from './ReviewChangesButton'
-import {IconBadge} from './IconBadge'
-import {
-  BadgesBox,
-  MetadataBox,
-  ReviewChangesBadgeBox,
-  ReviewChangesButtonBox,
-} from './DocumentSparkline.styled'
 import {PublishStatus} from './PublishStatus'
+import {ReviewChangesButton} from './ReviewChangesButton'
 
 export const DocumentSparkline = memo(function DocumentSparkline() {
-  const {editState, historyController, value} = useDocumentPane()
+  const {
+    editState,
+    historyController,
+    value,
+    documentId,
+    documentType,
+    handleHistoryClose,
+    handleHistoryOpen,
+    changesOpen,
+  } = useDocumentPane()
+  const syncState = useSyncState(documentId, documentType)
+
   const lastUpdated = value?._updatedAt
   const lastPublished = editState?.published?._updatedAt
   const showingRevision = historyController.onOlderRevision()
   const liveEdit = Boolean(editState?.liveEdit)
   const published = Boolean(editState?.published)
   const changed = Boolean(editState?.draft)
-  const loaded = published || changed
 
-  // Keep track of the size of the review changes button
-  const [
-    reviewChangesButtonElement,
-    setReviewChangesButtonElement,
-  ] = useState<HTMLButtonElement | null>(null)
-  const reviewChangesButtonElementRect = useElementRect(reviewChangesButtonElement)
-  const reviewChangesButtonWidth = reviewChangesButtonElementRect?.width
-  const reviewChangesButtonWidthRef = useRef(reviewChangesButtonWidth || 0)
+  const [rootFlexElement, setRootFlexElement] = useState<HTMLDivElement | null>(null)
+  const rootFlexRect = useElementRect(rootFlexElement)
+  const collapsed = Boolean(rootFlexRect && rootFlexRect?.width < 300)
```

Comment:
Here the `collapsed` value will be `false` when the `rootFlexRect` element doesn't exist. I personally think this is not "mobile-first". Would be great if this was `collapsed=true` until the element exists.

Something like:
```ts
const collapsed = !rootFlexRect || rootFlexRect.width < 300
```

Reply:
Good point 👍